cmd/kubectl-gadget/main.go

```diff
@@ -63,24 +59,20 @@ func main() {
 
 	common.AddVerboseFlag(rootCmd)
 
-	// grpcruntime.New() will try to fetch the info from the cluster by
-	// default. Make sure we don't do this when certain commands are run
-	// (as they just don't need it or imply that there are no nodes to
-	// contact, yet).
-	skipInfo := false
-	isHelp := false
-	isVersion := false
-	isDeployUndeploy := false
+	// Some commands don't need the gadget namespace. Run then before to avoid
+	// printing warnings about gadget namespace not found.
+	needGadgetNamespace := true
+
+	if len(os.Args) == 1 {
+		needGadgetNamespace = false
+	}
+
+	// Need to loop through all arguments to skip flags...
 	for _, arg := range os.Args[1:] {
-		for _, skipCmd := range infoSkipCommands {
-			if arg == skipCmd {
-				skipInfo = true
-			}
+		switch arg {
+		case "completion", "deploy", "version":
```

Comment:
```suggestion
		case "completion", "deploy", "version", "--help", "help", "-h":
```
There is now a difference in the help messages:
```
./kubectl-gadget                            
Collection of gadgets for Kubernetes developers

Usage:
  kubectl-gadget [command]
```
```
./kubectl-gadget --help
WARN[0000] No running Inspektor Gadget instances found. 
WARN[0000] Failed to load deploy info: loading gadget info from remote: dialing random target: get gadget pods: no gadget pods found in namespace "gadget". Is Inspektor Gadget deployed? 
Collection of gadgets for Kubernetes developers

Usage:
```

Reply:
But then the deploy info won't be loaded and the help output will be incomplete:

```
...
Usage:
  kubectl-gadget [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  deploy      Deploy Inspektor Gadget on the cluster
  help        Help about any command
  undeploy    Undeploy Inspektor Gadget from cluster
  version     Show version
...
```

Reply:
Corner cases after corner cases. This whole logic is cursed